test/test_oauth2.py

```diff
@@ -234,6 +234,40 @@ def test_it_creates_a_profile_when_exchanging(test_client: FlaskClient) -> None:
                   json={'access_token': '1/fFAGRNJru1FTz70BzhT3Zg', 'expires_in': 3920,
                         'foo': 'bar', 'token_type': 'Bearer', 'orcid': '0000-0002-1825-0097',
                         'name': 'Josiah Carberry'})
+    responses.add(responses.GET, 'http://www.example.com/api/v2.0/0000-0002-1825-0097/record',
```

Comment:
we are at the limit of what can be tested at the API level, a single email address should be fine; but more complex cases can be tested at the domain or (if the email update command is extracted) immediately over that, without involving the outer API and the token exchange

Reply:
Yeah, extracting `update_from_orcid_record()` would make such testing easier. And adding unit tests for `Profile`.

Reply:
Begun in https://github.com/elifesciences/profiles/pull/27.